### PR TITLE
feat(onedrive): return item info on request

### LINKED ISSUE #123: getDriveItemFromShareLink returns no details if link has format `/drives/...`
In OneDrive.getDriveItemFromShareLink:

https://github.com/adobe/helix-onedrive-support/blob/ce86668d97d6795c250438281a47f8d46e448872/src/OneDrive.js#L313-L317

if the link given starts with `/drives/...`, only very basic information is returned (`id` and `parentReference`), as opposed to when that link starts with `https://`.  I see no way to get more information on the item (e.g. its last modified date) in the former case.

I suggest to either retrieve the information before returning from `getDriveItemFromShareLink` or add an optional parameter (e.g. `fetchInfo = false`) that fills the missing information. That information is returned from: 
```
https://graph.microsoft.com/v1.0/drives/<drive-id>/items/<item-id>
```

This feature would particularly help in solving https://github.com/adobe/helix-data-embed/issues/233

### PR DESCRIPTION
**Pull request by dominique-pfister**
fix #123

### CODE REVIEW

File: src/OneDrive.d.ts [hunk 1]
```diff
@@ -129,9 +129,10 @@ export declare class OneDrive extends EventEmitter {
   /**
    * Returns a drive item from the given share link or onedrive uri.
    * @param {string|URL} url The share link url or a onedrive uri.
+   * @param {boolean} [fetchInfo=false] If {@code true} fetch detailed information about item
    * @see OneDrive.driveItemToURL
    */
-  getDriveItemFromShareLink(url: string|URL): Promise<DriveItem>;
+  getDriveItemFromShareLink(url: string|URL, fetchInfo?: boolean): Promise<DriveItem>;
```
Comment (tripodsan) [inline src/OneDrive.d.ts:135]: `fetchInfo` sounds weird... maybe `ensureQualified` or something

File: src/OneDrive.js [hunk 2]
```diff
@@ -310,9 +310,19 @@ class OneDrive extends EventEmitter {
 
   /**
    */
-  async getDriveItemFromShareLink(sharingUrl) {
+  async getDriveItemFromShareLink(sharingUrl, fetchInfo = false) {
     let driveItem = OneDrive.driveItemFromURL(sharingUrl);
     if (driveItem) {
+      if (fetchInfo) {
+        try {
```
Comment (tripodsan) [inline src/OneDrive.js:317]: why not use: `getDriveItem()` 
https://github.com/adobe/helix-onedrive-support/blob/ce86668d97d6795c250438281a47f8d46e448872/src/OneDrive.js#L429

File: src/OneDrive.js [hunk 3]
```diff
@@ -310,9 +310,19 @@ class OneDrive extends EventEmitter {
...
+          const uri = `/drives/${driveItem.parentReference.driveId}/items/${driveItem.id}`;
+          return await (await this.getClient()).get(uri);
```
Comment (tripodsan) [inline src/OneDrive.js:319]: the _normal_ sharelinks are cached, but not this ones....

**Review by tripodsan — CHANGES_REQUESTED** [review]
I don't really like this improvement. I think the user of the OneDrive api can use `getDriveItem()`, if there is not enough information.

but maybe the documentation of `getDriveItemFromShareLink` should be updated to note, that it might return a sparse drive item object.

File: src/OneDrive.js (hunk 2 above)
Comment (dominique-pfister) [reply src/OneDrive.js:317]: Ah, missed that one 🙄 I think I was confused by the parameter name `folderItem`, so I didn't realise this works for other items as well, thanks!

**Comment by dominique-pfister** [comment]
Verified approach with OneDrive.getDriveItem() as suggested.

**Pull request closed without merging**